### PR TITLE
LPD 38677 master

### PR DESCRIPTION
**Pull request by titokenzo**
- **LPD-38677 fixes test broken by LPD-23830 (commit a27a9d5268dd36ebe4731b64f517d987b8c95729), the backgoundTask returns an invalid companyId for the test, so a mock for companyId has been added. This also exposes a logical problem when restoring ThreadLocalValues with backgroundTask companyId**
- **LPD-38677 fixes regression from LPD-23830 (commit a27a9d5268dd36ebe4731b64f517d987b8c95729), backgroundTask companyId cannot be used to restore threadLocalValue, as it was used to change the original value**

In ThreadLocalAwareBackgroundTaskExecutorTest, since we mocked backgroundTask, we also needed to mock the calls for getTaskContextMap and getCompanyId.
Also added a method that generates mixed test values in BaseBackgroundTaskTestCase, so we verify if the values were restored after run the executor.

### CODE REVIEW

**Comment by titokenzo** [comment]
ci:test:sf

**Comment by titokenzo** [comment]
ci:test:relevant

**Comment by titokenzo** [comment]
ci:test:core-functional

File: modules/apps/portal-background-task/portal-background-task-service/src/test/java/com/liferay/portal/background/task/internal/BaseBackgroundTaskTestCase.java [hunk 1]
```diff
@@ -190,6 +190,22 @@ protected HashMap<String, Serializable> initializeThreadLocalValues() {
 		).build();
 	}
 
+	protected HashMap<String, Serializable> initializeThreadLocalValuesMixed() {
```
Comment (tinatian) [inline modules/apps/portal-background-task/portal-background-task-service/src/test/java/com/liferay/portal/background/task/internal/BaseBackgroundTaskTestCase.java:193]: @titokenzo I do not understand why you need to duplicate the logic in initializeThreadLocalValues

Comment (tinatian) [reply modules/apps/portal-background-task/portal-background-task-service/src/test/java/com/liferay/portal/background/task/internal/BaseBackgroundTaskTestCase.java:193]: If this is something only be used by specific test, then it should not be added to this base class

**Comment by tinatian** [comment]
@titokenzo Close as we talked, please resend. Thanks!

**Pull request closed without merging**